src/service/Websocket.ts

```diff
@@ -0,0 +1,66 @@
+// import { newPendingStateEvent } from '../foundation.js';
+import {
+  APPLICATION_ERROR,
+  extractErrorMessage,
+  parseXml,
+  SERVER_ERROR,
+} from './foundation.js';
+
+export function websocket(
+  element: Element,
+  serviceName: string,
+  url: string,
+  request: string
+): Promise<Document> {
+  let websocketInstance: WebSocket | undefined;
+
+  async function sleep(sleepTime: number): Promise<unknown> {
+    return setTimeout(() => {
+      console.log();
+    }, sleepTime);
+  }
+
+  async function waitUntilExecuted(): Promise<void> {
+    while (websocketInstance !== undefined) {
+      await sleep(250);
+    }
+  }
+
+  return new Promise<Document>((resolve, reject) => {
+    websocketInstance = new WebSocket(url);
+
+    websocketInstance.onopen = () => {
+      websocketInstance?.send(request);
```

Comment:
` websocketInstance! ` you can be certain this is not null, because it's in the onopen function of the same `websocketInstance`